### DISCUSSION TITLE
What we can do with breaking changes?

### DISCUSSION THREAD

**Discussion by ZeaLoVe** [databendlabs/databend#7268] [General]
With rapid iteration, some functional features may conflict with previous versions. These conflicts can affect users' usage. Is there a way to highlight this in the release note or weekly report? Also, can such a feature design be brought up for discussion earlier?

Especially SQL dialect changed, commands changed.

**Comment by flaneur2020** [comment]
I guess we can mark the changed behaviour with a semantic release manner: make a BREAKING: as prefix in the pull request.

and we need some introduction about why this behaviour is changed to users.

**Reply by ZeaLoVe** [reply]
Release note maybe not be enough, it just shows in the single release version, but breaking affects the following versions. Better a page to show all breaking infomation.
Like https://etcd.io/docs/v3.5/upgrades/upgrading-etcd/. If there is breaking, pick the version in upgrading pages, show all breaking changes on this page, and better some guides.

**Comment by Xuanwo** [comment]
Here are some experiences from [OpenDAL](https://github.com/datafuselabs/opendal)

OpenDAL doesn't reach 1.0, so OpenDAL can introduce breaking changes between minor versions. To make our users happier, OpenDAL maintains a [Upgrade](https://opendal.databend.rs/upgrade.html) docs. OpenDAL will record the changes and how to migrate from existing code for every breaking change. Sometimes, we will add links to RFC to explain why we introduce this change.

For example, OpenDAL 0.12 adds breaking changes, so we have the following section:

![image](https://user-images.githubusercontent.com/5351546/186834584-21a05dff-b7f9-41ce-8a36-dd631b1a18ac.png)

FYI: https://opendal.databend.rs/upgrade.html
